plugins/snowflake/snowsql.go

```diff
@@ -0,0 +1,32 @@
+package snowflake
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func SnowflakeCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Snowflake",
+		Runs:    []string{"snowsql"},
+		DocsURL: sdk.URL("https://docs.snowflake.com/en/user-guide/snowsql.html"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWhenContainsArgs("-a"),
+			needsauth.NotWhenContainsArgs("--accountname"),
+			needsauth.NotWhenContainsArgs("-u"),
+			needsauth.NotWhenContainsArgs("--username"),
```

Comment:
Is it possible that, if the command contains only one of these flags, the rest of the credentials could be retrieved from 1Password? 

In that sense, I'd like to only opt out of authentication when the command contains _all_ the necessary flags for specifying credentials. What do you think?

Reply:
What would be really awesome, perhaps we can open a ticket about it: When the user specifies `--accountname`, the CLI automatically provisions the credential that has the `account` field set to that value. Likewise for the other two credentials.

Reply:
I like that idea @hculea! Let's indeed open a follow-on ticket for that. While it's a great addition, there's already much value in including the Snowflake plugin before the credential selection through the `--account` flag is implemented.

Reply:
@hculea I like that suggestion as well! One thing I do want to mention is that there's no way to pass the password as an argument for the SnowSQL CLI, only account or username.

Reply:
@hculea Would you be able to create the new follow-up issue? Or if you would prefer, I would be happy to write it up too.
For now, I'm marking this thread as resolved!

Reply:
Thanks for expanding, William. In this case, what is the reason that we'd like to opt out of authentication when these flags are provided? The password would still need to be retrieved from 1Password.

Reply:
Simon - absolutely, I wouldn't like to hold this back, my proposal is something we can do as an iterative improvement if anything. I think there are many plugins that could benefit from such a callback system 😄

Reply:
From my understanding, this situation falls under the question: "Should we allow users to bypass the shell plugin, if they want to specify some credentials manually?" @arunsathiya proposed this approach in this [comment](https://github.com/1Password/shell-plugins/pull/161#discussion_r1084899549).

The way I see it, here are the use cases:
1. User does not have all their accounts' credentials stored in 1P, and they would like to authenticate with another set of credentials, or they just want to quickly use some demo/test creds for some testing, where security is not a concern. 

> In this case, what is the reason that we'd like to opt out of authentication when these flags are provided? 

When one of `--accountname` or `--username` is provided, if we choose to bypass shell plugin auth, the default behaviour of `snowsql` would be to interactively prompt the user for specifying all the missing credentials (such as `password`). 

If we were not to bypass shell plugin auth, when someone would specify `--accountname`, for instance, the arg value will take precedence over our provisioned  `SNOWSQL_ACCOUNT` env var. However, the other secrets (`username` and `password`) will come from our `SNOWSQL_USER` and `SNOWSQL_PWD` and this will result in a credentials mismatch most likely. In this case, providing args in the command would prove breaking.

For the `--config` flag things are simple because this would point `snowsql` to take the creds straight from a config file, and regardless whether we enforce shell plugin auth or not, the config file passed through the flag will take priority over other credentials.

Therefore, I would say in this case we have 2 options between we should decide:
- we let users bypass shell plugin auth if they want to use other credentials, and we choose to remove shell plugin auth if any of the flags `--accountname`,`--username` or `--config` are present (as @williamhpark is currently doing)
- we direct users to import all their credential sets into 1P, and then we fallback on use case 2.

2. User has all their credentials in 1Password but would like to dynamically choose which one to use. 

Users could currently achieve this by re-doing the step of `op plugin init snowsql` and choose the other item where they stored their credentials. 

@hculea proposed a cooler alternative: matching the flags `--accountname`,`--username` in the command with the credentials configured to be working with the plugin, and failing if nothing matches.

Reply:
> Therefore, I would say in this case we have 2 options between which we should decide:

What do you think @SimonBarendse @hculea @williamhpark  and @arunsathiya ?

Reply:
Hmm ok makes sense to let this be a way to opt out of the authentication.

We can afterwards iteratively add the dynamic matching of flags with 1password fields, and still support this behaviour, in case no creds are found. I'm okay to leave this as-is.

Reply:
Thank you for laying all of that out @AndyTitu! Makes perfect sense to me.

I agree that matching the passed flags with 1Password fields would be really cool, and would probably be super convenient for certain developers who cycle between several accounts often.

Reply:
Sounds good. With this in mind, I'll be resolving this thread.

I'll make a note to open an internal ticket based on our discussion above.